### PR TITLE
move local cluster import from ACM to MCE

### PR DESCRIPTION
**Pull request by ray-harris**
Signed-off-by: Ray Harris <raharris@redhat.com>

### CODE REVIEW

**Comment by JakobGray** [comment]
Tried an installation with this and MCE's build. Here are some issues I encountered:
- MCH is still installing after 10 minutes. The following is reported in its status, which I imagine is contributing to it not reaching Ready state
```
      local-cluster:
        lastTransitionTime: "2022-10-20T02:59:29Z"
        message: No conditions available
        reason: No conditions available
        status: Unknown
        type: Unknown
```
- I tried setting the local-cluster component to `enabled: false` in MCH to see if it would propagate to MCE and got the following error: `error: multiclusterhubs.operator.open-cluster-management.io "multiclusterhub" could not be patched: admission webhook "multiclusterhub.validating-webhook.open-cluster-management.io" denied the request: invalid component config: local-cluster is not a known component`

**Comment by ray-harris** [comment]
For the second one, in MCH we already had the spec.disableHubSelfManagement field. I kept using this as moving that to the component list would be a breaking change in the CRD.

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -553,6 +553,11 @@ func GetMCEComponents(mch *operatorsv1.MultiClusterHub) []mcev1.ComponentConfig
 			config = append(config, mcev1.ComponentConfig{Name: n, Enabled: mch.Enabled(n)})
 		}
 	}
+	if mch.Spec.DisableHubSelfManagement {
```
Comment (cameronmwall) [inline pkg/utils/utils.go:556]: Am I correct in reading this that there are now 2 places to disable hub self-management? Can we have the case where we have set `MCELocalCluster` `enabled` set to false without `disableHubSelfManagement` being set?

**Comment by JakobGray** [comment]
I did a test run and most things seem to be working. The MCH install was able to progress to completion. One important thing was missing regarding the Acceptance Criteria
> ACM continues to maintain KlusterletAddonConfig for now

When I do a `oc get KlusterletAddonConfig -A` it comes back with `No resources found`. It looks like this was removed from the ACM operator, but also not migrated to the MCE operator.

File: pkg/utils/utils.go (hunk 1 above)
Comment (ray-harris) [reply pkg/utils/utils.go:556]: When ACM is installed, `disableHubSelfManagement` will be used to control `MCELocalCluster`. If you change `MCELocalCluster` with ACM installed, mch-o should change it back. I realized I haven't tested this yet and will today.

**Pull request merged**